src/senaite/sync/browser/views.py

```diff
@@ -304,12 +309,35 @@ def update_object_with_data(self, obj, data, domain):
                     response = requests.get(url)
                     value = response.content
 
+            # Leave the Proxy Fields for later
+            if isinstance(fm, ProxyFieldManager):
+                proxy_fields.append({'field_name': fieldname,
+                                     'fm': fm, 'value': value})
+                continue
+
             logger.info("Setting value={} on field={} of object={}".format(
                 repr(value), fieldname, api.get_id(obj)))
             try:
                 fm.set(obj, value)
             except:
-                logger.error("Could not set field '{}' with value '{}'".format(fieldname, value))
+                logger.error(
+                    "Could not set field '{}' with value '{}'".format(
+                        fieldname, value))
+
+        # All reference fields are set. We can set the proxy fields now.
+        for pf in proxy_fields:
+            field_name = pf.get("field_name")
+            fm = pf.get("fm")
+            value = pf.get("value")
+            logger.info("Setting value={} on field={} of object={}".format(
+                repr(value), field_name, api.get_id(obj)))
+            try:
+                fm.set(obj, value)
+            except:
```

Comment:
Pease try to avoid blind excepts. Check the field manager to see what errors it throws (I guess `ValueError or `TypeError`)